lib/lz4.c

```diff
@@ -548,7 +548,13 @@ LZ4_FORCE_INLINE int LZ4_compress_generic(
 
                 match = LZ4_getPositionOnHash(h, cctx->hashTable, tableType, base);
                 if (dict==usingExtDict) {
-                    if (match < (const BYTE*)source) {
+                    if (match == 0) {
```

Comment:
OK, so according to this logic,
the code only looks for a match in the dictionary if it could not find one in the current data segment.
Well, I think it works.
For the speed / compression trade-off, we'll need a few test corpus to benchmark.